documentation/src/test/java/org/hibernate/search/documentation/mapper/orm/alternative/alternativebinder/AlternativeBinderIT.java

```diff
@@ -44,14 +45,12 @@ public static List<? extends Arguments> params() {
 	}
 
 	@RegisterExtension
-	public DocumentationSetupHelper setupHelper = DocumentationSetupHelper.withSingleBackend(
-			BackendConfigurations.simple() );
+	public DocumentationSetupHelper setupHelper = DocumentationSetupHelper.withSingleBackend( BackendConfigurations.simple() );
 	private EntityManagerFactory entityManagerFactory;
 
 	public void init(Boolean annotationProcessingEnabled, HibernateOrmSearchMappingConfigurer mappingContributor) {
-		setupHelper.withAnnotationProcessingEnabled( annotationProcessingEnabled )
-				.withMappingConfigurer( mappingContributor );
-		entityManagerFactory = setupHelper.start().setup( BlogEntry.class );
+		entityManagerFactory = setupHelper.start( variant( annotationProcessingEnabled, mappingContributor ) )
+				.setup( BlogEntry.class );
```

Comment:
Interestingly, this feels very close to `@ParameterizedSetup`, except... the setup needs to be re-executed for every test method (at least in this test, because we need to clean up data).

Maybe you could use `@ParameterizedSetup` more broadly if there was a way to tell it when to perform setup? E.g. `@ParameterizedSetup(PER_CLASS)` vs. `@ParameterizedSetup(PER_METHOD)`?

Reply:
see `AnalysisBuiltinIT` as an example 😃